tests/components/synology_dsm/test_init.py

```diff
@@ -0,0 +1,32 @@
+"""Tests for the Synology DSM component."""
+from homeassistant.components.synology_dsm import _async_setup_services
+from homeassistant.components.synology_dsm.const import DOMAIN, SERVICES
+from homeassistant.config_entries import SOURCE_USER
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_SSL,
+    CONF_USERNAME,
+)
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .consts import HOST, PASSWORD, PORT, USE_SSL, USERNAME
+
+
+async def test_services_registered(hass: HomeAssistantType):
+    """Test if all services are registered."""
+    await hass.config_entries.flow.async_init(
```

Comment:
We want to set up the integration here with `hass.config_entries.async_setup` and inactivate the default fixture and instead just patch the library, to avoid I/O.

Create a mock config entry with `MockConfigEntry` and add it it to hass first.
https://github.com/home-assistant/core/blob/239cc23b9b06c8a81012b8a30935a36dd4f8ad96/tests/common.py#L658

Reply:
I hope, I get it now 🤞 😃